### PR TITLE
Closes #1795 add assertions/stopifnot for widget inputs

### LINKED ISSUE #1795: QC: Add data structure checks of arguments to to `Widget_*` functions.
## QC Details
- `dfResults` is a data frame
- `lMetric` is list but not a data frame
- `dfGroups` is a data frame
- etc.

### PR DESCRIPTION
**Pull request by zdz2101**
## Overview
- `stopifnot()` chunk was added at the beginning of each widget app

## Connected Issues

- Closes #1795

### CODE REVIEW

File: tests/testthat/test_Widget_BarChart.R [hunk 1]
```diff
@@ -40,3 +40,60 @@ test_that("Widget_BarChart processes dfGRoups correctly", {
   reportingGroups_json <- jsonlite::toJSON(reportingGroups, na = "string")
   expect_equal(widget$x$dfGroups, reportingGroups_json)
 })
+
+test_that("Widget_BarChart assertions works", {
```
Comment (zdz2101) [inline tests/testthat/test_Widget_BarChart.R:44]: @samussiah did you want the unit test for the respective argument in that same `test_that` chunk? For now I just slotted them all as an "assertions" chunk at the end of each file

File: R/Widget_BarChart.R [hunk 2]
```diff
@@ -41,6 +41,16 @@ Widget_BarChart <- function(
   strShinyGroupSelectID = "GroupID",
   bDebug = FALSE
 ) {
+  stopifnot(
```
Comment (zdz2101) [inline R/Widget_BarChart.R:44]: That said I just followed the convention of the `stopifnot()` chunk that was in the `Analyze_xxx()` functions

Comment (zdz2101) [inline R/Widget_BarChart.R:44]: That said I just followed the convention of the `stopifnot()` chunk that was in the `Analyze_xxx()` functions

File: tests/testthat/test_Widget_BarChart.R (hunk 1 above)
Comment (samussiah) [reply tests/testthat/test_Widget_BarChart.R:44]: Makes sense to me!

File: R/Widget_BarChart.R [hunk 3]
```diff
@@ -41,6 +41,16 @@ Widget_BarChart <- function(
...
+    "dfResults is not a data.frame" = is.data.frame(dfResults),
+    "lMetric must be a list, but not a data.frame" = is.list(lMetric) & !is.data.frame(lMetric),
```
Comment (samussiah) [inline R/Widget_BarChart.R:46]:
```suggestion
    "lMetric must be a list, but not a data.frame" = is.list(lMetric) && !is.data.frame(lMetric),
```

File: R/Widget_BarChart.R [hunk 4]
```diff
@@ -41,6 +41,16 @@ Widget_BarChart <- function(
...
+    "dfGroups is not a data.frame" = is.null(dfGroups) | is.data.frame(dfGroups),
```
Comment (samussiah) [inline R/Widget_BarChart.R:47]:
```suggestion
    "dfGroups is not a data.frame" = is.null(dfGroups) || is.data.frame(dfGroups),
```

**Review by samussiah — APPROVED** [review]
Super useful @zdz2101!

**Pull request merged**